inst/include/Rcpp/exceptions.h

```diff
@@ -89,6 +89,48 @@ namespace Rcpp {
             file_io_error("file already exists", file) {}	// #nocov end
     };
 
+    #define RCPP_ADVANCED_EXCEPTION_CLASS(__CLASS__, __WHAT__)                                                                                                                                                     \
```

Comment:
I think we may have tried that too in the past and retreated.  Foggy memory on this though...